great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -278,7 +281,10 @@ def __init__(self, runtime_environment: Optional[dict] = None) -> None:
         )
         # Init stores
         self._stores: dict = {}
-        self._init_stores(self.project_config_with_variables_substituted.stores)
+        self._init_primary_stores(self.project_config_with_variables_substituted.stores)
+        # The DatasourceStore is inherent to all DataContexts but is not an explicit part of the project config.
```

Comment:
Do we know if there is a reason for this?

Reply:
We didn't want to make it a point of extensibility like the other stores - we added it as a way to interact with YAML in a clean way but users shouldn't interact with it